src/extension.ts

```diff
@@ -105,6 +105,25 @@ export function activate(context: ExtensionContext) {
 							lastStatus = item.text;
 							resolve();
 							commands.registerCommand(Commands.START_DEBUG_SESSION, async (config) => {
+								if (!config.request) { // if 'request' is missing interpret this as a missing launch.json
+									config.type = 'java';
+									config.name = 'Java Debug';
+									config.request = 'launch';
+									const editor = vscode.window.activeTextEditor;
+									if (editor && editor.document.languageId === 'java') {
+										const fullpath = editor.document.fileName;
+										const cur = path.parse(fullpath).dir;
+										config.startupClass = path.parse(fullpath).base.slice(0, -5);
+										config.sourcePath = [cur];
```

Comment:
when the single file defines the package name, and the config.sourcePath option you give will also contain the package path. Not sure if the DebugAdapter can infer the right file path when the breakpoint occurs. Did you test this case?